src/fidesops/service/privacy_request/request_service.py

```diff
@@ -0,0 +1,73 @@
+import logging
+from datetime import datetime
+from typing import Optional, Any, Dict, Set, List
+
+from sqlalchemy.orm import Session
+
+from fidesops.models.policy import Policy, ActionType
+from fidesops.models.privacy_request import PrivacyRequestStatus, PrivacyRequest
+from fidesops.schemas.drp_privacy_request import DrpPrivacyRequestCreate
+from fidesops.schemas.masking.masking_configuration import MaskingConfiguration
+from fidesops.schemas.masking.masking_secrets import MaskingSecretCache
+from fidesops.schemas.policy import Rule
+from fidesops.schemas.redis_cache import PrivacyRequestIdentity
+from fidesops.service.masking.strategy.masking_strategy_factory import get_strategy
+
+logger = logging.getLogger(__name__)
+
+
+def retrieve_policy(db: Session, field: str, value: Any) -> Optional[Policy]:
+    return Policy.get_by(
+        db=db,
+        field=field,
+        value=value,
+    )
+
+
+def build_required_privacy_request_kwargs(
+    requested_at: Optional[datetime], policy_id: str
+) -> Dict[str, Any]:
+    return {
+        "requested_at": requested_at,
+        "policy_id": policy_id,
+        "status": PrivacyRequestStatus.pending.value,
+    }
+
+
+def create_privacy_request(db: Session, kwargs: Dict[str, Any]):
+    return PrivacyRequest.create(db=db, data=kwargs)
+
+
+def cache_data(
+    privacy_request: PrivacyRequest,
+    policy: Policy,
+    identity: PrivacyRequestIdentity,
+    encryption_key: Optional[str],
+    drp_request_body: Optional[DrpPrivacyRequestCreate],
+) -> None:
+    # Store identity and encryption key in the cache
+    logger.info(f"Caching identity for privacy request {privacy_request.id}")
+    privacy_request.cache_identity(identity)
+    privacy_request.cache_encryption(encryption_key)  # handles None already
+
+    # Store masking secrets in the cache
+    logger.info(f"Caching masking secrets for privacy request {privacy_request.id}")
+    erasure_rules: List[Rule] = policy.get_rules_for_action(
+        action_type=ActionType.erasure
+    )
+    unique_masking_strategies_by_name: Set[str] = set()
+    for rule in erasure_rules:
+        strategy_name: str = rule.masking_strategy["strategy"]
+        configuration: MaskingConfiguration = rule.masking_strategy["configuration"]
+        if strategy_name in unique_masking_strategies_by_name:
+            continue
+        unique_masking_strategies_by_name.add(strategy_name)
+        masking_strategy = get_strategy(strategy_name, configuration)
+        if masking_strategy.secrets_required():
+            masking_secrets: List[
+                MaskingSecretCache
+            ] = masking_strategy.generate_secrets_for_cache()
+            for masking_secret in masking_secrets:
+                privacy_request.cache_masking_secret(masking_secret)
+    if drp_request_body:
+        privacy_request.cache_drp_request_body(drp_request_body)
```

Comment:
not sure why GH is showing weird indents here
<img width="581" alt="Screen Shot 2022-05-05 at 2 27 03 PM" src="https://user-images.githubusercontent.com/7697292/167010923-7758c3a4-d545-492a-9a7c-4b27a6641763.png">

Reply:
Looks the same to me, might be viewport width?

Reply:
yeah probably